crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -102,16 +143,28 @@ pub fn derive_bundle(input: TokenStream) -> TokenStream {
     let mut field_component_ids = Vec::new();
     let mut field_get_components = Vec::new();
     let mut field_from_components = Vec::new();
-    for (field_type, field) in field_type.iter().zip(field.iter()) {
-        field_component_ids.push(quote! {
-        <#field_type as #ecs_path::bundle::Bundle>::component_ids(components, storages, &mut *ids);
-        });
-        field_get_components.push(quote! {
-            self.#field.get_components(&mut *func);
-        });
-        field_from_components.push(quote! {
-            #field: <#field_type as #ecs_path::bundle::Bundle>::from_components(ctx, &mut *func),
-        });
+    for ((field_type, field_kind), field) in
+        field_type.iter().zip(field_kind.iter()).zip(field.iter())
+    {
+        match field_kind {
+            BundleFieldKind::Component => {
+                field_component_ids.push(quote! {
+                <#field_type as #ecs_path::bundle::Bundle>::component_ids(components, storages, &mut *ids);
+                });
+                field_get_components.push(quote! {
+                    self.#field.get_components(&mut *func);
+                });
+                field_from_components.push(quote! {
+                    #field: <#field_type as #ecs_path::bundle::Bundle>::from_components(ctx, &mut *func),
+                });
+            }
+
+            BundleFieldKind::Ignore => {
+                field_from_components.push(quote! {
+                    #field: Default::default(),
```

Comment:
I believe, to keep macros _hygienic_, you want to fully qualify any `std::` types/traits
 ```suggestion
                     #field: ::std::default::Default::default(),
```
I know it looks ridiculous, but it can save headache in the future.